sql/dd/impl/upgrade/server.cc

```diff
@@ -470,6 +470,67 @@ static void create_upgrade_file() {
 
 }  // namespace
 
+#ifdef WITH_WSREP
+
+bool pxc_fix_mysql_tables(THD *thd) {
+  const char section_start[] = "#! PXC_SECTION::START";
+  const char section_end[] = "#! PXC_SECTION::END";
+  int section_start_len = strlen(section_start);
+  int section_end_len = strlen(section_end);
+
+  const char **query_ptr;
+  bool in_pxc_section = false;
+
+  if (ignore_error_and_execute(thd, "USE mysql")) {
```

Comment:
Shouldn't we switch back to the current db at the end?

Reply:
as above, we're following the same conventions as the upgrade code (see fix_mysql_tables()).